ultralytics/solutions/parking_management.py

```diff
@@ -4,238 +4,168 @@
 from ultralytics.utils.plotting import Annotator
 
 
 class ParkingPtsSelection:
     """Class for selecting and managing parking zone points on images using a Tkinter-based UI."""
 
     def __init__(self):
-        """Initializes the UI for selecting parking zone points in a tkinter window."""
+        """Class initialization method."""
         check_requirements("tkinter")
+        import tkinter as tk
+        from tkinter import filedialog, messagebox
 
-        import tkinter as tk  # scope for multi-environment compatibility
+        self.tk, self.filedialog, self.messagebox = tk, filedialog, messagebox
+        self.setup_ui()
+        self.initialize_properties()
+        self.master.mainloop()
 
-        self.tk = tk
-        self.master = tk.Tk()
+    def setup_ui(self):
+        """Sets up the Tkinter UI components."""
+        self.master = self.tk.Tk()
         self.master.title("Ultralytics Parking Zones Points Selector")
-
-        # Disable window resizing
         self.master.resizable(False, False)
 
-        # Setup canvas for image display
+        # Canvas for image display
         self.canvas = self.tk.Canvas(self.master, bg="white")
+        self.canvas.pack(side=self.tk.BOTTOM)
 
-        # Setup buttons
+        # Button frame with buttons
         button_frame = self.tk.Frame(self.master)
         button_frame.pack(side=self.tk.TOP)
-
-        self.tk.Button(button_frame, text="Upload Image", command=self.upload_image).grid(row=0, column=0)
-        self.tk.Button(button_frame, text="Remove Last BBox", command=self.remove_last_bounding_box).grid(
-            row=0, column=1
-        )
-        self.tk.Button(button_frame, text="Save", command=self.save_to_json).grid(row=0, column=2)
-
-        # Initialize properties
-        self.image_path = None
-        self.image = None
-        self.canvas_image = None
-        self.rg_data = []  # region coordinates
-        self.current_box = []
-        self.imgw = 0  # image width
-        self.imgh = 0  # image height
-
-        # Constants
-        self.canvas_max_width = 1280
-        self.canvas_max_height = 720
-
-        self.master.mainloop()
+        for text, cmd in [
+            ("Upload Image", self.upload_image),
+            ("Remove Last BBox", self.remove_last_bounding_box),
+            ("Save", self.save_to_json),
+        ]:
+            self.tk.Button(button_frame, text=text, command=cmd).pack(side=self.tk.LEFT)
+
+    def initialize_properties(self):
+        """Initialize the necessary properties."""
+        self.image = self.canvas_image = None
+        self.rg_data, self.current_box = [], []
+        self.imgw = self.imgh = 0
+        self.canvas_max_width, self.canvas_max_height = 1280, 720
 
     def upload_image(self):
-        """Upload an image and resize it to fit canvas."""
-        from tkinter import filedialog
-
-        from PIL import Image, ImageTk  # scope because ImageTk requires tkinter package
```

Comment:
@RizwanMunawar I had this here for a reason and after you removed it now our Docker images are crashing.

Reply:
@glenn-jocher opened the PR for this fix: https://github.com/ultralytics/ultralytics/pull/16995